packages/react-reconciler/src/ReactFiberPerformanceTrack.js

```diff
@@ -507,36 +543,82 @@ export function logTransitionStart(
           : updateTime > 0
             ? updateTime
             : renderStartTime;
-      console.timeStamp(
-        eventIsRepeat ? '' : 'Event: ' + eventType,
-        eventTime,
-        endTime,
-        currentTrack,
-        LANES_TRACK_GROUP,
-        color,
-      );
+      if (__DEV__ && debugTask) {
+        debugTask.run(
+          // $FlowFixMe[method-unbinding]
+          console.timeStamp.bind(
+            console,
+            eventIsRepeat ? '' : 'Event: ' + eventType,
+            eventTime,
+            endTime,
+            currentTrack,
+            LANES_TRACK_GROUP,
+            color,
+          ),
+        );
+      } else {
+        console.timeStamp(
+          eventIsRepeat ? '' : 'Event: ' + eventType,
+          eventTime,
+          endTime,
+          currentTrack,
+          LANES_TRACK_GROUP,
+          color,
+        );
+      }
     }
     if (startTime > 0) {
       // Log the time from when we started an async transition until we called setState or started rendering.
-      console.timeStamp(
-        'Action',
-        startTime,
-        updateTime > 0 ? updateTime : renderStartTime,
-        currentTrack,
-        LANES_TRACK_GROUP,
-        'primary-dark',
-      );
+      // TODO: Ideally this would use the debugTask of the startTransition call perhaps.
```

Comment:
Right now it's using the one from the `setState` call that was wrapped in `startTransition`, right?

If I had to choose, I'd usually prefer the deeper one. Does `createTask` include the async stack? If not, you could set a breakpoint at the `setState` and find the corresponding `startTransition` that way if it's async. If the transition function is sync, having the deeper `setState` task is already perfect IMO.

The task for `startTransition` could leave you hanging if the transition function has multiple `setState` (maybe even in separate if-else).

Reply:
Yea. I was thinking about maybe starting to log the Event and Action entries inside the `setState` call itself instead of waiting for the render to start. That way it would have the stack trace of the `setState` even in the profiling builds which would not have `createTask` enabled if the `console.timeStamp` call itself had it natively. The downside is that the "Update" track would not have it since it gets logged at the end time.

However, we can also just use `createTask` in profiling builds just for this case since it's just once per `setState` and cheap when DevTools is not open.